### PR TITLE
Fix missing connectionTerminate handling

### PR DESCRIPTION
**Pull request by harryyu1994**
- A patch for #6430 Inform JITServer when JITClient closes connections
  - #6430 caused a double free bug which is fixed by this change
- A few rename changes

[skip ci]

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

File: runtime/compiler/control/JITaaSCompilationThread.cpp [hunk 1]
```diff
@@ -4408,20 +4408,25 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
...
+            }
+         else if (e.getType() == JITServer::MessageType::clientTerminate)
+            {
+            deleteClientSessionData(e.getClientId(), compInfo, compThread);
+            // Delete server stream
+            stream->~ServerStream();
+            TR_Memory::jitPersistentFree(stream);
+            entry._stream = NULL;
+            }
+         // else e.getType() == JITServer::MessageType::compilationAbort
```
Comment (mpirvu) [inline runtime/compiler/control/JITaaSCompilationThread.cpp]: The compilationAbort case is already handled above

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8 depends eclipse/openj9-omr#jitaas

File: runtime/compiler/control/JITaaSCompilationThread.cpp (hunk 1 above)
Comment (harryyu1994) [reply runtime/compiler/control/JITaaSCompilationThread.cpp]: Once the `enableJITaaSPerCompConn` is removed, we can put the three of them together to make this look more clean.

**Comment by harryyu1994** [comment]
Once this one and https://github.com/eclipse/openj9/pull/6610 are merged, https://github.com/eclipse/openj9/pull/6629 can go into `master`.

**Pull request merged**